tests/rptest/tests/franz_go_verifiable_test.py

```diff
@@ -83,3 +86,123 @@ def test_with_all_type_of_loads(self):
                                         start_timeout=200,
                                         stop_timeout=200)
             time.sleep(30)
+
+
+class SiWithFranzGoTest(RedpandaTest):
```

Comment:
I would vote for a more descriptive name to replace `FranzGoTest`. The name could reflect the goals of that test harness rather than the name of its underlying client. Maybe something like `Verifier`-something assuming the intention is to replace our existing verifier based around the java client.